### PR TITLE
fix: boards: update m5stack_cores3 configuration

### LINKED ISSUE #81496: build/twister fail: m5stack_cores3/esp32s3/appcpu samples/synchronization
**Describe the bug**

build & twister fail of samples/synchronization fails on m5stack_cores3/esp32s3/appcpu

**To Reproduce**

west build -p -b m5stack_cores3/esp32s3/appcpu zephyr/samples/synchronization -T sample.kernel.synchronization  

west twister -p m5stack_cores3/esp32s3/appcpu -s samples/synchronization/sample.kernel.synchronization

**Expected behavior**

Build should succeed

**Impact**

Annoyance.
CI failure

**Logs and console output**

### PR DESCRIPTION
**Pull request by sylvioalves**
This board has a few issues handled by this PR:
- Fix DTS entries to meet necessary flash partitions
- Fix wrong kconfig entries related to SoC model
- Fix documentation

This fixes/updates [76914](https://github.com/zephyrproject-rtos/zephyr/pull/76914), which was old and not rebased as necessary, causing CI issues.

Fixes #81496

### CODE REVIEW

File: boards/m5stack/m5stack_cores3/Kconfig.m5stack_cores3 [hunk 1]
```diff
@@ -4,6 +4,6 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config BOARD_M5STACK_CORES3
-	select SOC_ESP32S3
+	select SOC_ESP32S3_WROOM_N16R8
```
[the remark this answers is not in the document]
Comment (marekmatej) [reply boards/m5stack/m5stack_cores3/Kconfig.m5stack_cores3:7]: The configuration is not the same, the SOC/Module configuration is defined here https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/espressif/esp32s3/Kconfig.soc and it reflects mostly the size of flash and spiram available on the module.

**Review by marekmatej — DISMISSED** [review]
LGTM

**Review by kartben — APPROVED** [review]

File: boards/m5stack/m5stack_cores3/Kconfig.m5stack_cores3 (hunk 1 above)
[the remark this answers is not in the document]
Comment (marekmatej) [reply boards/m5stack/m5stack_cores3/Kconfig.m5stack_cores3]: @soburi I've realized the M5stack_cores3 is not based on any of ESP32S3 modules so it should not use any of the module configurations as you correctly stated.

**Review by marekmatej — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

File: boards/m5stack/m5stack_cores3/Kconfig.m5stack_cores3 (hunk 1 above)
[the remark this answers is not in the document]
Comment (soburi) [reply boards/m5stack/m5stack_cores3/Kconfig.m5stack_cores3]: Thank you for your confirmation.

**Review by soburi — APPROVED** [review]

**Pull request merged**